extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/TransitKeyConfig.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.vault.runtime.config;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class TransitKeyConfig {
+
+    /**
+     * Specifies the name of the key to use. By default this will be the property key alias. Used when
+     * the same transit key is used with different configurations. Such as:
+     * 
+     * <pre>
+     *     {@code
+     *     quarkus.vault.transit.key.foo1.name=foo
+     *     quarkus.vault.transit.key.foo1.key_version=1
+     *     quarkus.vault.transit.key.foo2.name=foo
+     *     quarkus.vault.transit.key.foo2.key_version=2
+     *     }
+     * </pre>
+     */
+    @ConfigItem
+    public Optional<String> name;
```

Comment:
How can this `name` property be used ? I think it can indeed be interesting for example to inject, for example, an already decrypted value as a property, in which case one would srart from setting a key name in the configuration.
Or indeed we can have this property set, but then there should be the options to call the transit engine without a key name.
Should we discuss now how it can be done, or postpone ?

Reply:
it is there to support 2 use cases:

the first one allows using a name in quarkus that is different than the real key name in vault:

```
quarkus.vault.transit.key.my-key.name=this_is_my_real_key
```

the second one is about creating different key alias with different configurations but based on the same vault key.

for instance I could have one key called 'my-key', and specify:

```
quarkus.vault.transit.key.my-prehashed-key.prehashed=true
```

so when I have prehashed data, I use `my-prehashed-key`, and `my-key` when I have the whole input.

I guess you took this as default key to use, which is not the case.
I actually thought about that use case very early on, but that would require to increase the number of methods, such as:
```
String encrypt(String clearData);
```
and I think we have too many methods already.

Reply:
@vsevel Sounds good, thanks